server/store/setting_store.go

```diff
@@ -61,22 +63,51 @@ func (s *pluginStore) GetSetting(userID, settingID string) (interface{}, error)
 	}
 }
 
-func (s *pluginStore) getDailySummarySettingForUser(userID string) (string, error) {
+func (s *pluginStore) getDailySummarySettingForUser(userID string) (*DailySummarySettings, error) {
 	dsumIndex, err := s.LoadDailySummaryIndex()
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
 	for _, dsum := range dsumIndex {
 		if dsum.MattermostUserID == userID {
-			if !dsum.Enable {
-				return "Daily summary not set", nil
+			return dsum, nil
+		}
+	}
+
+	return nil, nil
+}
+
+func (s *pluginStore) updateDailySummarySettingForUser(userID string, value interface{}) error {
+	dsumIndex, err := s.LoadDailySummaryIndex()
+	if err != nil {
+		return err
+	}
+
+	for _, dsum := range dsumIndex {
+		if dsum.MattermostUserID == userID {
+			stringValue := value.(string)
+			if stringValue == "true" {
+				dsum.Enable = true
+				break
 			}
-			return fmt.Sprintf("Daily summary set at %s (%s)", dsum.PostTime, dsum.Timezone), nil
+
+			if stringValue == "false" {
+				dsum.Enable = false
+				break
+			}
+
+			dsum.PostTime = value.(string)
+			break
 		}
 	}
 
-	return "Daily summary not set", nil
+	err = s.SaveDailySummaryIndex(dsumIndex)
```

Comment:
`StoreDailySummaryUserSettings` is available so you don't need to access the index directly